plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/ssl/ReloadableSecureAuxTransportSslContext.java

```diff
@@ -0,0 +1,134 @@
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSessionContext;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+
+import io.grpc.netty.shaded.io.netty.buffer.ByteBufAllocator;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolNegotiator;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+
+/**
+ * A reloadable implementation of io.grpc.SslContext.
+ * This object delegates its functionality to an SSLEngine as retrieved from SecureAuxTransportSettingsProvider::buildSecureAuxServerEngine.
+ * Delegating functionality to buildSecureAuxServerEngine take immediate advantage of cert/key updates.
+ * This additionally means our SSLEngine is volatile and may change between any two function calls.
+ * In the worst case the information provided during the initial handshake such as supported ciphers is stale by the time
+ * we create and return the newEngine, resulting in a failed connection.
+ */
+public class ReloadableSecureAuxTransportSslContext extends SslContext {
+    private static final String[] DEFAULT_SSL_PROTOCOLS = { "TLSv1.3", "TLSv1.2", "TLSv1.1" };
+    private static final String[] HTTP2_ALPN = { "h2" };
+    private final SecureAuxTransportSettingsProvider provider;
+    private final boolean isClient;
+
+    /**
+     * Create and SSLEngine from the default provider if one is set.
+     * @return default javax.net.ssl.SSLEngine instance.
+     */
+    private SSLEngine getDefaultServerSSLEngine() {
+        try {
+            final SSLEngine engine = SSLContext.getDefault().createSSLEngine();
+            engine.setEnabledProtocols(DEFAULT_SSL_PROTOCOLS);
+            return engine;
+        } catch (final NoSuchAlgorithmException ex) {
+            throw new OpenSearchSecurityException("Unable to initialize default server SSL engine", ex);
+        }
+    }
+
+    /**
+     * Initializes a new ReloadableSecureAuxTransportSslContext.
+     * @param provider creation of new SSLEngine instances is delegated to this object.
+     * @param isClient determines if handshake is negotiated in client or server mode.
+     */
+    public ReloadableSecureAuxTransportSslContext(SecureAuxTransportSettingsProvider provider, boolean isClient) {
+        super();
+        this.provider = provider;
+        this.isClient = isClient;
+    }
+
+    /**
+     * Create a new SSLEngine instance to handle TLS for a connection.
+     * @param byteBufAllocator provider interface does not allow us to leverage an allocator so this param is not used.
+     * @return new SSLEngine instance.
+     */
+    @Override
+    public SSLEngine newEngine(ByteBufAllocator byteBufAllocator) {
+        SSLEngine sslEngine;
+        try {
+            sslEngine = provider.buildSecureAuxServerEngine().orElseGet(this::getDefaultServerSSLEngine);
```

Comment:
It seems like the `ReloadableSecureAuxTransportSslContext` supposed to support client as well?

```suggestion
           if (isClient == true) {
                sslEngine = provider.buildSecureAuxClientEngine().orElseGet(this::getDefaultServerSSLEngine);|
            }
```

Reply:
It depends if we would provide the GRPC client configuration (for plugins, fe?)